Python/pylifecycle.c

```diff
@@ -262,30 +262,86 @@ initexternalimport(PyInterpreterState *interp)
 int
 _Py_LegacyLocaleDetected(void)
 {
+    /* This API is expected to be called before the Python runtime is
+     * initialised, so it needs to be very conservative as to which other APIs
+     * it calls, and which process resources it attempts to access.
+     */
 #ifndef MS_WINDOWS
-    /* On non-Windows systems, the C locale is considered a legacy locale */
-    /* XXX (ncoghlan): some platforms (notably Mac OS X) don't appear to treat
-     *                 the POSIX locale as a simple alias for the C locale, so
-     *                 we may also want to check for that explicitly.
+    /* On non-Windows systems, the ASCII-based C locale is considered a legacy
+     * locale that we don't trust to be accurate.
+     * Some platforms (notably Mac OS X and other *BSD platforms) don't treat
+     * the POSIX locale as a simple alias for the C locale, so we also check
+     * for that explicitly.
      */
     const char *ctype_loc = setlocale(LC_CTYPE, NULL);
-    return ctype_loc != NULL && strcmp(ctype_loc, "C") == 0;
+    return ctype_loc != NULL &&
+           (strcmp(ctype_loc, "C") == 0 || strcmp(ctype_loc, "POSIX") == 0);
 #else
     /* Windows uses code pages instead of locales, so no locale is legacy */
     return 0;
 #endif
 }
 
+int
+_Py_LegacyLocaleCoercionEnabled(int argc, char **argv)
+{
+    int enable_coercion = -1;
+    if (argc >= 1 && argv != NULL) {
+        /* Rudimentary arg parsing to allow -E and -I to disable
```

Comment:
I was hoping to get away without needing to rely on the full `getopt` argument definition string here, but realised that's a futile endeavour due to the fact that cases like `python -iEV` are perfectly valid option strings.

Reply:
Not so futile after all - with all the cross-platform getopt code migrated to `wchar_t`, it proved simpler to keep this early pass looking specifically for `-E` and `-I` independent of it.

Even adding a `-X no_locale_coercion` option would be straightforward (but I've chosen not to implement that at this stage)